src/main/com/fulcrologic/rad/database_adapters/datomic_cloud.clj

```diff
@@ -0,0 +1,648 @@
+(ns com.fulcrologic.rad.database-adapters.datomic-cloud
+  (:require
+    [clojure.spec.alpha :as s]
+    [clojure.set :as set]
+    [clojure.walk :as walk]
+    [clojure.pprint :refer [pprint]]
+    [com.fulcrologic.fulcro.algorithms.tempid :as tempid]
+    [com.fulcrologic.fulcro.algorithms.do-not-use :refer [deep-merge]]
+    [com.fulcrologic.guardrails.core :refer [>defn => ?]]
+    [com.fulcrologic.rad.attributes :as attr]
+    [com.fulcrologic.rad.authorization :as auth]
+    [com.fulcrologic.rad.form :as form]
+    [com.fulcrologic.rad.ids :refer [select-keys-in-ns]]
+    [com.rpl.specter :as sp]
+    [com.wsscode.pathom.connect :as pc]
+    [com.wsscode.pathom.core :as p]
+    [datomic.client.api :as d]
+    [edn-query-language.core :as eql]
+    [taoensso.encore :as enc]
+    [compute.datomic-client-memdb.core :as memdb]
+    [taoensso.timbre :as log])
+  (:import (java.util UUID)))
+
+(def type-map
```

Comment:
A *lot* of code duplication in this ns...why not move a lot of this stuff to a Datomic common ns.